app/models/concerns/profile_tailoring.rb

```diff
@@ -58,4 +58,10 @@ def update_os_minor_version(version)
 
     update!(os_minor_version: version)
   end
+
+  def value_overrides_by_ref_id
+    ValueDefinition.where(id: value_overrides.keys).each_with_object({}) do |value_definition, overrides|
+      overrides[value_definition.ref_id] = value_overrides[value_definition.id]
+    end
+  end
```

Comment:
Does this look better? @skateman

Reply:
Yup, just please test it :)

Reply:
I tested and it does only generates the one `ValueDefinition Load` query